src/SprykerEco/Yves/Computop/Mapper/Init/PostPlace/DirectDebitMapper.php

```diff
@@ -0,0 +1,141 @@
+
+        $computopPaymentTransfer->setCapture(ComputopSharedConfig::CAPTURE_MANUAL_TYPE);
+        $computopPaymentTransfer->setTransId($this->generateTransId($quoteTransfer));
+        $computopPaymentTransfer->setMandateId($computopPaymentTransfer->getTransId());
+        $computopPaymentTransfer->setUrlSuccess(
+            $this->getAbsoluteUrl($this->application->path(ComputopControllerProvider::DIRECT_DEBIT_SUCCESS))
+        );
+        $computopPaymentTransfer->setOrderDesc(
+            $this->computopService->getTestModeDescriptionValue($quoteTransfer->getItems()->getArrayCopy())
+        );
+
+        return $computopPaymentTransfer;
+    }
+
+    /**
+     * @param \Spryker\Shared\Kernel\Transfer\TransferInterface $cardPaymentTransfer
+     *
+     * @return array
+     */
+    protected function getDataSubArray(TransferInterface $cardPaymentTransfer)
+    {
+        /** @var \Generated\Shared\Transfer\ComputopDirectDebitPaymentTransfer $cardPaymentTransfer */
+        $dataSubArray[ComputopApiConfig::TRANS_ID] = $cardPaymentTransfer->getTransId();
+        $dataSubArray[ComputopApiConfig::AMOUNT] = $cardPaymentTransfer->getAmount();
+        $dataSubArray[ComputopApiConfig::CURRENCY] = $cardPaymentTransfer->getCurrency();
+        $dataSubArray[ComputopApiConfig::URL_SUCCESS] = $cardPaymentTransfer->getUrlSuccess();
+        $dataSubArray[ComputopApiConfig::URL_FAILURE] = $cardPaymentTransfer->getUrlFailure();
+        $dataSubArray[ComputopApiConfig::CAPTURE] = $cardPaymentTransfer->getCapture();
+        $dataSubArray[ComputopApiConfig::RESPONSE] = $cardPaymentTransfer->getResponse();
+        $dataSubArray[ComputopApiConfig::MAC] = $cardPaymentTransfer->getMac();
+        $dataSubArray[ComputopApiConfig::ORDER_DESC] = $cardPaymentTransfer->getOrderDesc();
+        $dataSubArray[ComputopApiConfig::ETI_ID] = ComputopConfig::ETI_ID;
+        $dataSubArray[ComputopApiConfig::MANDATE_ID] = $cardPaymentTransfer->getMandateId();
+        $dataSubArray[ComputopApiConfig::DATE_OF_SIGNATURE_ID] = $this->getDateOfSignature();
+
+        return $dataSubArray;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDateOfSignature()
+    {
+        $now = new DateTime();
+
+        return $now->format(self::DATE_FORMAT);
+    }
+
+    /**
+     * @return string
+     */
+    protected function getActionUrl()
+    {
+        return $this->config->getDirectDebitInitAction();
+    }
+
+    /**
+     * @param string $merchantId
+     * @param string $data
+     * @param int $length
+     *
+     * @return string
+     */
+    protected function getUrlToComputop($merchantId, $data, $length)
+    {
+        $queryData = [
+            ComputopApiConfig::MERCHANT_ID => $merchantId,
+            ComputopApiConfig::DATA => $data,
+            ComputopApiConfig::LENGTH => $length,
+        ];
+
+        return $this->getActionUrl() . '?' . http_build_query($queryData);
```

Comment:
this is dangerous, since this call is hidden.

I would create functions:
1. getQueryParameters - building array queryData
2. getActionUrl - receiving parameters list and building complet url.